src/fetch-wrapper.ts

```diff
@@ -104,14 +108,14 @@ export default function fetchWrapper(requestOptions: RequestOptions) {
             url,
             status,
             headers,
-            data: await getResponseData(requestOptions, response),
+            data: parseResponse ? await getResponseData(response) : response.body,
```

Comment:
I don't think we should apply response parsing in case of errors or 304 status, only for 2xx responses.

Reply:
Actually when status is 304 usually dont have payload. But i dont know about githubs api. 304 should imho be returned if we send e.g. the etag or not modified since in the request header. And if we match that, it should just send 304 without payload. But it seems that the api is expecting that we maybe have payload. So it could be, that we send 304 on the tarball and would get the file we requested. 

Also it seems arbitrary to sometimes handle parseResponse. Imho i would apply it always where we call getResponseData.

Reply:
304 is a special case used for conditional requests. You are right there is definitely no body in the response in case of a 304, but it's outside of the scope of this PR.

> Also it seems arbitrary to sometimes handle parseResponse. Imho i would apply it always where we call getResponseData.

In this example

```
try {
  const requestOptions = request('GET /repos/{owner}/{repo}/tarball/{ref}', {
    owner: 'OWNER',
    repo: 'REPO',
    ref: 'REF'
    request: { parseResponse: false }
  })
} catch (error) {
  // handle error
}
```

When I handle the error, I'd expect and very much prefer for error.response.body to be set to the object that is returned from GitHub's API which explains the error, and not a stream I'd need to stream and parse first. Error responses are very small, there is no need to stream it.

Reply:
indeed, makes sense.